tools/config_generator.py

```diff
@@ -79,11 +95,123 @@ def update_param(config_dict, arg_dict):
             if arg_dict.get(key) is not None:
                 config_dict.update({key: arg_dict.get(key)})
 
-    if config_dict.get("general.random_seed") == "*":
-        del config_dict["general.random_seed"]
+    if config_dict.get("general.random_seed") == "#":
+        config_dict.pop("general.random_seed")
+
+    return config_dict
+
+
+def update_dataset_stats(dataset, config_dict):
+    dataset_stats = pd.read_csv(DATASET_STATS, sep='\t', header=None)
+    index = dataset_stats.index[dataset_stats[0] == dataset]
+    if (len(index) != 0):
+        stats = []
+        indices = [1, 3, 4, 5, 6]
+        for i in indices:
+            stats.append(dataset_stats.loc[index[0]].tolist()[i])
+        config_dict = update_stats(stats, config_dict)
+    else:
+        raise RuntimeError("Unrecognized dataset")
+
+    return config_dict
+
+
+def update_stats(stats, config_dict):
+    config_dict.update({"num_train": str(stats[2])})
+    config_dict.update({"num_nodes": str(stats[0])})
+    config_dict.update({"num_relations": str(stats[1])})
+    config_dict.update({"num_valid": str(stats[3])})
+    config_dict.update({"num_test": str(stats[4])})
 
     return config_dict
 
 
+def set_args():
+    parser = argparse.ArgumentParser(
+                description='Generate configs', prog='config_generator',
+                formatter_class=argparse.RawTextHelpFormatter,
+                epilog=('Specify certain config (optional): ' +
```

Comment:
Can you add a command which will print all possible options and the defaults? 

Printing the full list of options might be too much so maybe we can have it print the section names + descriptions with the following commands: `python3 config_generator.py --list_config_options` and/or `python3 config_generator.py -l`.

Then to print the options and defaults for a specific section we can do:  `python3 config_generator.py -l <section>`

To print all options: `python3 config_generator.py -l all`

Use https://github.com/marius-team/marius/blob/main/docs/configuration.rst for reference with this

Reply:
If this change is too large we can make a separate PR for it.

Reply:
Will make a separate PR for this.